### DISCUSSION TITLE
Unhandled Migration Error Avalanche C Chain

### DISCUSSION THREAD

**Discussion by 7FigureSwagger** [ConsenSys-archive/truffle#5645] [General Q&A]
Also posted on [ethereum.stackexchange](https://ethereum.stackexchange.com/questions/138157/truffle-migration-error-avalanche-c-chain)

I am using Truffle suite in a project in production, but I am sick of dealing with this error I feel shouldn't even be happening, and maybe its because of something that Avalanche has set on their side, but I am hoping for some clarity here.

I keep getting presented with this error when running migrations, it happens at entirely random intervals (mid script after 4 successful txs, while also happening immidiately when the script is run). This error happens every migration, run of a script, or anything targeting C-Chain (mainnet). Even a script calling 'read-only' functions...

> Error: Unhandled error. ({ code: -32000, message: 'cannot query unfinalized data' })
    at new NodeError (node:internal/errors:371:5)
    at Web3ProviderEngine.emit (node:events:379:17)

I have asked this question already in the Avalanche dev discord chat, and I was informed verbatim that: 

> That error message is returned when someone specifies a height that is > than the last accepted block height.

Not only this, but allegedly this same check is present on fuji testnet, and i NEVER EVER EVER have this issue on fuji testnet...

Since I am using truffle's contract abstractions, I am not aware how I could be 'specifying' anything, in fact I added the Avalanche recommended gas levels to the truffle-config, which did not help at all. There is no mention of this error in truffle docs anywhere or this concept from what i can find. Not as it relates to errors in migrations or with the truffle contracts package etc...

PS: Problem does NOT rear its ugly head when I perform migrations on Moonbeam, or Ethereum

**Comment by apdan510** [comment]
Hey @7FigureSwagger sorry for the delay. This could possibly be a bug. Any chance you can share the repo with us?
